strax/storage/mongo.py

```diff
@@ -115,6 +126,27 @@ def _build_chunk_registry(self, backend_key):
             # issues or json-encoding headaches.
             self.chunks_registry[backend_key + str(chunk_key)] = doc.copy()
 
+        # Some bookkeeping to make sure we don't buffer too much in this
+        # backend. We still need to return at least one hence the 'and'.
+        # See: https://github.com/AxFoundation/strax/issues/346
+        if backend_key not in self._buffered_backend_keys:
+            self._buffered_backend_keys.append(backend_key)
+        while ((getsizeof(self.chunks_registry) / 1e6 > self._buff_mb
+                and len(self._buffered_backend_keys) > 1)
+               or len(self._buffered_backend_keys) > self._buff_nruns):
+            self._clean_first_key_from_registry()
+
+    def _clean_first_key_from_registry(self):
+        """
+        Remove the first item in the self.buffered_keys and all the
+        associated keys in the self.chunks_registry to limit RAM-usage
+        """
+        # only clean the first entry from the list
+        to_clean = self._buffered_backend_keys[0]
+        for registry_key in list(self.chunks_registry.keys()):
+            if to_clean in registry_key:
+                del self.chunks_registry[registry_key]
```

Comment:
Python's complained at me if I modified a dictionary while iterating over it. List-casting the keys probably helps with this, but I got around it by storing the keys to delete in a second list.

Reply:
you really have a good memory for errors, got to give you that. This is true. 

However if you look carefully we are making a copy of the keys by iterating them when converting to a list. Therefore you are not iterating over the keys but over the list of strings that are extracted